packages/core/src/controllers/W3aCtrl.ts

```diff
@@ -0,0 +1,105 @@
+import { proxy, ref, subscribe as valtioSub } from 'valtio/vanilla'
+import type { W3aCtrlState } from '../types/controllerTypes'
+import { IFRAME_API, IFRAME_ID, IS_SERVER, call, messageHandler } from '../utils/W3aUtil'
+
+// -- initial state ------------------------------------------------ //
+const state = proxy<W3aCtrlState>({
+  iFrameMsgIndex: 0,
+  iFrameReady: false,
+  providerInited: false,
+  iFrame: null as unknown as HTMLIFrameElement,
```

Comment:
Ui should be completely different from the controllers i.e. think about controllers being portable to other platforms like react-native.